pkg/util/deadlock-detector.go

```diff
@@ -55,17 +53,35 @@ func DeadlockWatchdogReadLock(lock *sync.RWMutex, name string, maxLockPeriod tim
 	DeadlockWatchdog(&rwMutexToLockableAdapter{lock}, name, maxLockPeriod)
 }
 
-func DeadlockWatchdog(lock Lockable, name string, maxLockPeriod time.Duration) {
+func DeadlockWatchdog(lock sync.Locker, name string, maxLockPeriod time.Duration) {
 	detector := &deadlockDetector{
-		lock:  lock,
-		name:  name,
-		clock: RealClock{},
+		lock:          lock,
+		name:          name,
+		exitChannelFn: func() <-chan time.Time { return time.After(maxLockPeriod) },
 	}
 	go detector.run()
 }
 
+// Useful for injecting tests
+type exiter interface {
+	Exitf(format string, args ...string)
+}
+
+type realExiter struct{}
+
+func (realExiter) Exitf(format string, args ...interface{}) {
```

Comment:
Wait, I guess this is the one that doesn't match the interface. But the interface is wrong.